lib/puppet/util.rb

```diff
@@ -313,7 +313,7 @@ def absolute_path?(path, platform=nil)
               raise Puppet::DevError, _("unknown platform %{platform} in absolute_path") % { platform: platform }
             end
 
-    !! (path =~ regex)
+    !! (path.to_s =~ regex)
```

Comment:
Nick fixed something similar in b0b6410bc659977382a2ca7fc232802d0c7c31b4. Maybe keep this method as is so we require a String, and fix AutosignSetting so it only calls `absolute_path?` if the value is a String?

Reply:
This is called in quite a few places in addition to AutosignSetting

```
lib/puppet/file_serving/base.rb:    Puppet::Util.absolute_path?(path, :posix) || (Puppet::Util::Platform.windows? && Puppet::Util.absolute_path?(path, :windows))
lib/puppet/file_serving/fileset.rb:    raise ArgumentError.new(_("Fileset paths must be fully qualified: %{path}") % { path: path }) unless Puppet::Util.absolute_path?(path)
lib/puppet/file_serving/terminus_selector.rb:    if Puppet::Util.absolute_path?(request.key)
lib/puppet/indirector/exec.rb:    raise ArgumentError, _("You must set the exec parameter to a fully qualified command") unless Puppet::Util.absolute_path?(external_command[0])
lib/puppet/indirector/request.rb:      if key.to_s =~ /^\w+:\// and not Puppet::Util.absolute_path?(key.to_s) # it's a URI
lib/puppet/parameter/path.rb:    fail(_("%{name} must be a fully qualified path") % { name: name }) unless Array(paths).all? {|path| absolute_path?(path)}
lib/puppet/parser/files.rb:    if Puppet::Util.absolute_path?(reference)
lib/puppet/parser/files.rb:    if path == "" || Puppet::Util.absolute_path?(path)
lib/puppet/parser/functions/generate.rb:      raise Puppet::ParseError, _("Generators must be fully qualified") unless Puppet::Util.absolute_path?(args[0])
lib/puppet/provider/exec.rb:    self.fail _("'%{exe}' is not qualified and no path was specified. Please qualify the command or specify a path.") % { exe: exe } if !absolute_path?(exe) and resource[:path].nil?
lib/puppet/provider/exec/windows.rb:    if absolute_path?(exe)
lib/puppet/provider/package/gem.rb:        if Puppet::Util.absolute_path?(source)
lib/puppet/settings/autosign_setting.rb:    elsif Puppet::Util.absolute_path?(value)
lib/puppet/type/file.rb:      unless Puppet::Util.absolute_path?(value)
lib/puppet/type/file.rb:      unless Puppet::Util.absolute_path?(value)
lib/puppet/type/file/source.rb:        next if Puppet::Util.absolute_path?(source)
lib/puppet/type/file/source.rb:        if Puppet::Util.absolute_path?(source)
lib/puppet/type/filebucket.rb:        if value.is_a? String and not Puppet::Util.absolute_path?(value)
lib/puppet/type/package.rb:      if source && absolute_path?(source)
lib/puppet/type/user.rb:            valid_home = Puppet::Util.absolute_path?(entry) || entry =~ %r{^~/|^%h/}
lib/puppet/util.rb:    if absolute_path?(bin)
lib/puppet/util.rb:  def absolute_path?(path, platform=nil)
lib/puppet/util.rb:  module_function :absolute_path?
lib/puppet/util/autoload.rb:      if Puppet::Util.absolute_path?(path)
lib/puppet/util/autoload.rb:    raise ArgumentError, _("Autoload paths cannot be fully qualified") if Puppet::Util.absolute_path?(@path)
lib/puppet/util/log/destinations.rb:    obj.is_a?(String) && Puppet::Util.absolute_path?(obj)
spec/unit/util_spec.rb:  describe "#absolute_path?" do
```

If we don't fix it here, the rest of these calls will also throw the deprecation warning. I feel like it's better to fix this here? Rather than this check, I was thinking we could use `return False unless path.is_a?(String)`